Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -739,17 +717,16 @@ AppDelegateViewModelOutputs {
     return applicationDidFinishLaunchingReturnValueProperty.value
   }
 
-  fileprivate let notificationAuthorizationCompletedProperty = MutableProperty()
-  public func notificationAuthorizationCompleted() {
-    self.notificationAuthorizationCompletedProperty.value = ()
+  fileprivate let notificationAuthorizationCompletedProperty = MutableProperty(false)
+  public func notificationAuthorizationCompleted(isGranted: Bool) {
+    self.notificationAuthorizationCompletedProperty.value = isGranted
   }
 
   fileprivate let notificationAuthorizationStatusProperty =
     MutableProperty<NotificationAuthorizationStatusType?>(nil)
-  public func notificationAuthorizationStatusReceived(
-    _ notificationSetting: NotificationAuthorizationStatusType
-  ) {
-    return self.notificationAuthorizationStatusProperty.value = notificationSetting
+  @available(iOS 10.0, *)
+  public func notificationAuthorizationStatusReceived(_ authorizationStatus: UNAuthorizationStatus) {
+    return self.notificationAuthorizationStatusProperty.value = authStatusType(for: authorizationStatus)
```

Comment:
I don't think anything wrong with using the `authStatusType` function, but I have general question:
If I would be implementing this I would probably think "'NotificationAuthorizationStatus' is the replacement type for the 'UNAuthorizationStatus' and for that I should have a factory static method:"
```
extension NotificationAuthorizationStatus {
    static func createFrom(uiAuthStatus: UNAuthorizationStatus) -> NotificationAuthorizationStatus
}
```
I there specific opinion/approach in our code where we would prefer one way to another?